### PR TITLE
fix(rust-libp2p): extend required checks

### PR DESCRIPTION
**Pull request by mxinden**
### Summary

Extend the list of required GitHub CI actions for the rust-libp2p respository.

### Why do you need this?

In order for mergify to not merge pull requests with failing CI.

### What else do we need to know?

**DRI:** myself

### Reviewer's Checklist

- [ ] It is clear where the request is coming from (if unsure, ask)
- [ ] All the automated checks passed
- [ ] The YAML changes reflect the summary of the request
- [ ] The Terraform plan posted as a comment reflects the summary of the request

### CODE REVIEW

**Comment by mxinden** [comment]
Let's wait for @thomaseizinger review here. I am not sure some of these were left out deliberately.

**Review by galargh — APPROVED** [review]

File: github/libp2p.yml [hunk 1]
```diff
@@ -6962,11 +6965,14 @@ repositories:
             - Compile on wasm32-wasi
             - Compile on x86_64-apple-darwin
             - Compile on x86_64-pc-windows-msvc
+            - Compile with MSRV
             - Compile with select features (mdns tcp dns async-std)
             - Compile with select features (mdns tcp dns tokio)
             - Ensure that `Cargo.lock` is up-to-date
             - examples
+            - gather_published_crates
```
Comment (thomaseizinger) [inline github/libp2p.yml:6973]: This is a helper job, that we may rename / change so lets remove this one.


```suggestion
```

File: github/libp2p.yml [hunk 2]
```diff
@@ -286,11 +286,14 @@ repositories:
         - tbenbow
     default_branch: main
     delete_branch_on_merge: false
+    files:
+      .github/workflows/stale.yml:
+        content: .github/workflows/stale.yml
```
Comment (thomaseizinger) [inline github/libp2p.yml:291]: This is already included further down below as an inline file. Why here again?

File: github/libp2p.yml [hunk 3]
```diff
@@ -286,11 +286,14 @@ repositories:
...
     has_discussions: false
     merge_commit_message: PR_TITLE
     merge_commit_title: MERGE_MESSAGE
-    secret_scanning_push_protection: false
-    secret_scanning: false
+    secret_scanning_push_protection: true
+    secret_scanning: true
```
Comment (thomaseizinger) [inline github/libp2p.yml:296]: Seems unrelated to the PR.

**Review by thomaseizinger — CHANGES_REQUESTED** [review]

File: github/libp2p.yml (hunk 2 above)
Comment (galargh) [reply github/libp2p.yml]: It's for a different repo. Resolved here: https://github.com/libp2p/github-mgmt/pull/180

File: github/libp2p.yml (hunk 3 above)
Comment (galargh) [reply github/libp2p.yml]: It is unrelated. It was automatically committed via "fix rules". It's been resolved separately in https://github.com/libp2p/github-mgmt/pull/180 and now these changes were removed from the PR.

**Review by thomaseizinger — APPROVED** [review]
